src/components/films/components/filmsList/FilmsList.tsx

```diff
@@ -2,44 +2,82 @@ import React, {useCallback, useEffect, useState} from "react";
 import FilmItemC from "./FilmItem";
 import { ContainerColumn, ContainerRowAlignStart } from  '../../../../styles/general';
 import { FilmsQ } from './filmsList.styled';
-import axios from "axios";
+import { useDispatch, useSelector } from "react-redux";
+import { getMoviesDataStart } from "./store/actions";
+import {IMovie} from "../../../../store/interfaces";
 
 export const FilmsListC = () => {
-  const [q, setQ] = useState<number>(1);
+  const [q, setQ] = useState<number>(0);
   const [moviesArr, setMoviesArr] = useState<[] | null>(null);
+  const dispatch = useDispatch();
+
+  // not sure about this ..
+  const movies = useSelector(state => state.moviesReducer.movies);
+  const filter = useSelector(state => state.filterReducer.filter);
+  const sortBy = useSelector(state => state.filterReducer.sortBy);
+
+  useEffect(() => {
+    dispatch(getMoviesDataStart());
+  }, []);
+
+  useEffect(() => {
+    movies && movies.length > 0 ? setMoviesArr(movies) : null;
+  }, [movies]);
 
   const getMoviesQ = useCallback(() => {
-    !!moviesArr ? setQ(moviesArr.length) : null;
+   !!moviesArr ? setQ(moviesArr.length) : null;
+  }, [moviesArr])
+
+
+  useEffect(() => {
+    getMoviesQ();
   }, [moviesArr])
 
   useEffect(() => {
-    axios.get(`http://localhost:9000/`)
-      .then(res => {
-        const movies = res.data;
+    if (filter && filter.length > 0) {
+      const filteredArr = movies.filter((el: IMovie) => el.category === filter);
+      setMoviesArr(filteredArr);
+
+      if (filter === 'all') {
         setMoviesArr(movies);
-      }).catch((err) => {
-        alert('There is a problem during fetching the data from database: '+ err)
-      })
+      }
+    }
+  }, [filter]);
 
-    getMoviesQ();
-  }, [getMoviesQ]);
+  Array.prototype.sortByF = function(p) {
```

Comment:
That might be considered an extremely [bad smell](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain#1_New_initialization)